Data/DataModel/SinglePlayerPuzzleUnlockState.cs

```diff
@@ -0,0 +1,24 @@
+﻿using System;
+using System.ComponentModel.DataAnnotations.Schema;
+
+namespace ServerCore.DataModel
+{
+    /// <summary>
+    /// The unlock state of a single player puzzle.
+    /// Unlike puzzles for teams, which allows admins/authors to unlock the puzzle for a specific team,
+    /// they will only be able to either lock or unlock for ALL players.
```

Comment:
How will we test these puzzles then?

Reply:
Will we treat them as standard team puzzles and switch them over late in the process?

Reply:
So. I was thinking that we would lock the puzzle behind a pre-requisite and we can mark as solved the prerequisite to do this. A little hacky but better than constantly checking for every player that logs in.

Reply:
Another way of doing it is to add a button that manually adds a player to the SinglePlayerPuzzleStatePerPlayer and allowing unlocks the typical way. Which I actually like more actually

Reply:
If the way I suggested is less work then I think it's fine as all puzzles will start their lives that way.

Reply:
sounds good to me!